scss/_badge.scss

```diff
@@ -1,42 +1,85 @@
-// Base class
 //
-// Requires one of the contextual, color modifier classes for `color` and
-// `background-color`.
+// Base styles
+//
 
 .badge {
   // scss-docs-start badge-css-vars
-  --#{$prefix}badge-padding: #{subtract(var(--#{$prefix}badge-padding-y), 1px) var(--#{$prefix}badge-padding-x) add(var(--#{$prefix}badge-padding-y), 1px)}; // OUDS mod
-  --#{$prefix}badge-padding-x: #{$badge-padding-x};
-  --#{$prefix}badge-padding-y: #{$badge-padding-y};
-  @include rfs($badge-font-size, --#{$prefix}badge-font-size);
-  --#{$prefix}badge-font-weight: #{$badge-font-weight};
-  --#{$prefix}badge-color: #{$badge-color};
-  --#{$prefix}badge-border-radius: #{$badge-border-radius};
+  --#{$prefix}badge-size-width: #{$ouds-badge-size-medium};
+  --#{$prefix}badge-size-min-width: #{$ouds-badge-size-medium};
+  --#{$prefix}badge-size-max-width: #{$ouds-badge-size-medium};
+  --#{$prefix}badge-size-height: #{$ouds-badge-size-medium};
+  --#{$prefix}badge-size-min-height: #{$ouds-badge-size-medium};
+  --#{$prefix}badge-size-max-height: #{$ouds-badge-size-medium};
+  --#{$prefix}badge-border-radius: #{$ouds-border-radius-pill};
+  --#{$prefix}badge-count-padding-x: #{$ouds-badge-space-inset};
+  --#{$prefix}badge-count-padding-y: #{$ouds-badge-space-inset};
   // scss-docs-end badge-css-vars
 
   display: inline-block;
-  padding: var(--#{$prefix}badge-padding);
-  @include font-size(var(--#{$prefix}badge-font-size));
-  font-weight: var(--#{$prefix}badge-font-weight);
-  line-height: 1;
-  color: var(--#{$prefix}badge-color);
+  width: var(--#{$prefix}badge-size-width);
+  min-width: var(--#{$prefix}badge-size-min-width);
+  max-width: var(--#{$prefix}badge-size-max-width);
+  height: var(--#{$prefix}badge-size-height);
+  min-height: var(--#{$prefix}badge-size-min-height);
+  max-height: var(--#{$prefix}badge-size-max-height);
+  padding: var(--#{$prefix}badge-count-padding-y) var(--#{$prefix}badge-count-padding-x);
+  font-style: normal;
+  font-weight: 400;
+  color: #{$ouds-color-content-on-status-neutral-emphasized};
   text-align: center;
-
-  /* rtl:remove */
-  letter-spacing: initial; // OUDS mod
-  white-space: nowrap;
-  vertical-align: baseline;
+  background-color: #{$ouds-color-surface-status-neutral-emphasized};
+  @include get-font-size("label-small");
   @include border-radius(var(--#{$prefix}badge-border-radius));
-  @include gradient-bg();
 
-  // Empty badges collapse automatically
-  &:empty {
-    display: none;
+  svg {
+    display: flex;
+    flex: 1 0 0;
+    aspect-ratio: 1/1;
+  }
+
+  &.badge-disabled {
+    color: #{$ouds-color-content-on-action-disabled};
+    background-color: #{$ouds-color-action-disabled};
+  }
+
+  &.badge-xs {
```

Comment:
Wondering if we can separate these classes to lesser the CSS specificity and ease the customization